### ISSUE TITLE
Protection of BCR if Magisk looses root (during call recording)

### ISSUE THREAD

**Issue by ElsAr4e** [chenxiaolong/BCR#585]
I've seen this issue: https://github.com/topjohnwu/Magisk/issues/8177

What will happen if you use BCR and root is suddenly lost (and maybe after some time root access works and then does not work again etc.)

- if you keep BCR running but you are not recording a call
- during a call recording of a regular call or emergency call

Can you make it technically possible that BCR will display a warning that you currently can't (continue to) record the call and/or tries to continue call recording (same call) once root access works again?

**Comment by chenxiaolong** [comment]
BCR will not be impacted by that at all because it does not rely on actual root access.

The Magisk module is only used as an easy method for installation as a system app. BCR only requires system app privileges.

**Comment by chenxiaolong** [comment]
Yeah, modules are handled by a different part of Magisk. Modules are loaded once during boot and as part of that, Magisk will mount files to the correct locations (eg. putting BCR's apk in `/system/priv-app`). Even if the Magisk app is broken or uninstalled, modules should still work properly.

The Magisk app is only needed for granting root access and installing/updating/removing modules, but modules that are already installed generally don't depend on the app.

**Comment by ElsAr4e** [comment]
@chenxiaolong Good to know. Thank you for your answer and explanation.
